test-harness/src/test/groovy/com/netflix/conductor/test/integration/SubWorkflowSpec.groovy

```diff
@@ -408,6 +415,9 @@ class SubWorkflowSpec extends AbstractSpecification {
             tasks[1].status == Task.Status.FAILED
         }
 
+        and:
```

Comment:
same as above, parent seems to be updated synchronously. Please verify if this is a side-effect.